core/types/transaction.go

```diff
@@ -42,9 +42,10 @@ var (
 
 // Transaction types.
 const (
-	LegacyTxType = iota
-	AccessListTxType
-	DynamicFeeTxType
+	LegacyTxType     = 0x00
+	AccessListTxType = 0x01
+	DynamicFeeTxType = 0x02
+	BlobTxType       = 0x05
```

Comment:
This is super weird, can we please have 0x04 and have L2s fix their shit?

Reply:
What's 0x03 though?

Reply:
I think they used non-eip-155 txs, so they didn't include the chain id. Or something werid like that. IMHO that's up on them then.

Reply:
Hmmm, it's really bad design IMHO. But you are right, it's up to them anyway. This transaction type definition shared by all different networks is a bit ugly I really want to say.

Reply:
Was fixed to be 0x03.